gel/_internal/_qb/_expressions.py

```diff
@@ -300,6 +300,35 @@ def __edgeql_expr__(self, *, ctx: ScopeContext) -> str:
         return f"<{self.type.as_quoted_schema_name()}>{expr}"
 
 
+@dataclass(kw_only=True, frozen=True)
+class ObjectWhenType(UnaryOp):
+    type_filter: TypeNameExpr
+
+    def __init__(
+        self,
+        *,
+        expr: ExprCompatible,
+        type_filter: TypeNameExpr,
+        type_: TypeNameExpr,
+    ) -> None:
+        op = _edgeql.Token.RANGBRACKET
+        object.__setattr__(self, "type_filter", type_filter)
```

Comment:
This is an issue for another PR (or not bothering), but all these __setattr__s and explicit `__init__`s in a dataclass are dodgy; are they all just so that `op` doesn't show up in the dataclass string like it would if it was declared with a default